### DISCUSSION TITLE
Bug in https://course.spacy.io/en/chapter1

### DISCUSSION THREAD

**Discussion by gitclem** [explosion/spaCy#7551] [Help: Other Questions] [answered]
On URL: https://course.spacy.io/en/chapter1

In section 12 "Writing match patterns", part 2, my solution is marked correct, but I expected THREE results not just TWO.  The automatic grader thinks TWO is correct.  It appears that `Winzip` isn't recognized as a `PROPN`.

This problem is probably due to the course being based on Spacy 2.x and not the newer Spacy 3.0

My solution:
```
import spacy
from spacy.matcher import Matcher

nlp = spacy.load("en_core_web_sm")
matcher = Matcher(nlp.vocab)

doc = nlp(
    "i downloaded Fortnite on my laptop and can't open the game at all. Help? "
    "so when I was downloading Minecraft, I got the Windows version where it "
    "is the '.zip' folder and I used the default program to unpack it... do "
    "I also need to download Winzip?"
)

# Write a pattern that matches a form of "download" plus proper noun
pattern = [{"LEMMA": "download"}, {"POS": "PROPN"}]

# Add the pattern to the matcher and apply the matcher to the doc
matcher.add("DOWNLOAD_THINGS_PATTERN", None, pattern)
matches = matcher(doc)
print("Total matches found:", len(matches))

# Iterate over the matches and print the span text
for match_id, start, end in matches:
    print("Match found:", doc[start:end].text)
```

OUTPUT:

```
Total matches found: 2
Match found: downloaded Fortnite
Match found: downloading Minecraft
✔ Good job!
```

When I tried it on my system with Spacy 3.0, I get this output:

```
Total matches found: 3
Match found: downloaded Fortnite
Match found: downloading Minecraft
Match found: download Winzip

```

**Comment by polm** [comment, marked as answer]
This is not really a bug, it's just an issue with the accuracy of the POS tagger. Some degree of errors with the models are inevitable; see [this issue](https://github.com/explosion/spaCy/issues/3052) for some notes on that. This case is a bit unfortunate but as you note it seems it's been fixed in v3.

Your answer is marked correct because your code is consistent with spaCy's predictions, which is all the answer-checking code has access to.
